lib/tests/sources.sh

```diff
@@ -25,12 +25,9 @@ cd "$work"
 
 touch {README.md,module.o,foo.bar}
 
-# nix-instantiate doesn't write out the source, only computing the hash, so
-# this uses the experimental nix command instead.
-
-dir="$(nix eval --impure --raw --expr '(with import <nixpkgs/lib>; "${
+dir="$(nix-instantiate --eval --strict --read-write-mode --json --expr '(with import <nixpkgs/lib>; "${
   cleanSource ./.
-}")')"
+}")' | cut -d \" -f 2)"
```

Comment:
I suppose this is a crude quote removal trick?
Please define a function for it, so that we can read this.

```suggestion
}")' | crudeUnquote)"
```

Reply:
I guess I can just add `jq` as a dependency, so I can do `jq -r` instead here

Reply:
sgtm. I think I've done that before and while it's not entirely "correctly typed",  it seems to work quite well.